src/api/app/models/user.rb

```diff
@@ -404,6 +404,12 @@ def is_staff?
     @is_staff = roles.exists?(title: 'Staff')
   end
 
+  def is_moderator?
+    return @is_moderator unless @is_moderator.nil?
+
+    @is_moderator = roles.exists?(title: 'Moderator')
```

Comment:
```suggestion
    @is_moderator ||= roles.exists?(title: 'Moderator')
```
Would this be the same?